options/options.go

```diff
@@ -18,7 +18,8 @@ type ProvenanceOpts struct {
 	// ExpectedSourceURI is the expected source URI in the provenance.
 	ExpectedSourceURI string
 
-	// ExpectedBuilderID is the expected builder ID.
+	// ExpectedBuilderID is the expected builder ID that is parsed from the envelope
+	// of the provenance.
```

Comment:
The builder ID is parsed from the envelope?

IIUC, This ID is also the one that the user expects and passes via the command line, not one that is parsed out of the provenance.

Reply:
You understand correctly. I think confused it when I wrote that comment. I was trying to differentiate between the two since they had the same comment, and `provenanceOpts.ExpectedBuilderID` could default to the delegator workflow when the `builderOpts.ExpectedBuilderID` is empty or nil.

Here is where `provenanceOpts.ExpectedBuilderID` gets set:
https://github.com/slsa-framework/slsa-verifier/blob/main/verifiers/internal/gha/verifier.go#L58C1-L71C1

That function [`VerifyBuilderIdentity`](https://github.com/slsa-framework/slsa-verifier/blob/main/verifiers/internal/gha/builder.go#L66) gets the builderID that sets `provenanceOpts.ExpectedBuilderID` [here](https://github.com/slsa-framework/slsa-verifier/blob/main/verifiers/internal/gha/builder.go#L86) which can default to the delegator in this conditional [here](https://github.com/slsa-framework/slsa-verifier/blob/main/verifiers/internal/gha/builder.go#L108C1-L110C3)

Perhaps a better comment would be:

`// ExpectedBuilderID is the expected builder ID that is passed from user and verified`

While for `BuilderOpts.ExpectedID' we could change the comment to be:

`// ExpectedBuilderID is the builderID passed in from the user to be verified`

Thoughts?

I think it is important to clear up the two because it can be confusing if reading the code for the first time.

Reply:
Changed comment.